app/models/query_execution.rb

```diff
@@ -24,21 +51,36 @@ def self.perform(result_id, role)
       rrrc = result.redis_result_row_count
 
       stream_processor = PgStream::Processor.new(query_stream)
-      stream_processor.register(ResultCsvGenerator.new(result_id, result.headers).callbacks)
+      stream_processor.register(ResultCsvGenerator.new(result.id, result.headers).callbacks)
       stream_processor.register(SampleSkimmer.new(NUM_SAMPLE_ROWS, &sample_callback).callbacks)
       stream_processor.register(CountPublisher.new(rrrc).callbacks)
 
       row_count = stream_processor.execute
       result.mark_complete_with_count(row_count)
     end
+
   rescue *RedshiftPG::USER_ERROR_CLASSES => e
     csv_service.clear_tmp_file
     result.mark_failed!(e.message)
-  rescue => e
-    if result && csv_service
-      csv_service.clear_tmp_file
-      result.mark_failed!(e.message)
+  end
+
+  def self.query_snowflake(connection, body, result, sample_callback)
+    # unload the query result from snowflake directly into s3
+    # then read in the first 100 rows from the file as sample rows
+    # Note: snowflake unload currently has a max file size of 5 GB.
+    connection.reconnect_on_failure do
+      location = File.join(connection.unload_target, result.current_result_filename)
+      sql = SNOWFLAKE_UNLOAD_SQL % {location: location, query: body, max_file_size: connection.max_file_size}
```

Comment:
_what kind of python sorcery is this?!_

Reply:
the kind from ruby amateur

Reply:
That sounds like a blanket term for all developers who write Ruby code.

Reply:
jokes aside, while i figured the template/interpolation operation was possible in ruby, i don't use it much, and didn't realize it was so similar to python's